### DISCUSSION TITLE
Creating foundation FEM in FreeCAD

### DISCUSSION THREAD

**Discussion by ebrahimraeyat** [xcfem/xc#120] [Show and tell]
Hello everyone and @lcpt . I am pleased to announce that I am working on creating foundation model for xc in my [OSAFE](https://github.com/ebrahimraeyat/OSAFE/tree/xc) FreeCAD workbench. 

![image](https://user-images.githubusercontent.com/8196112/230496115-d4513294-709d-4e27-893a-a7197d23dd6a.png)

![xc1](https://user-images.githubusercontent.com/8196112/230495983-d27a98f0-40ca-4085-bab4-742fa1d63b1e.png)

I have some questions, but I don't know if this is a suitable place for asking my question. I think it is all about xc. but it can be asked in FreeCAD FEM forum. Thanks.

**Comment by lcpt** [comment]
What are those questions?

**Comment by ebrahimraeyat** [comment]
Thanks @lcpt.
1- How can I add nodes into surfaces mesh. I mean column point loads
2- If some surfaces overlap each other, How GMesh meshes this areas? how can I find which nodes belongs to which surface?
3- It is common to define columns as stiff elements, as asked in question 2, how can I found which nodes belongs to stiff geometry?

Then I have some additional questions about the design of foundation. Now in CSI SAFE we define strips and safe reports moments and shears alongside of strip. I studied a little about it and found that it uses wood-armer theory:

[a sample link](https://www.linkedin.com/pulse/what-wood-armer-method-concrete-slab-design-chanthoeun-chiv/)

Do you have information in this regards? or have any example?

Thanks again.

**Comment by lcpt** [comment]
Hi, @ebrahimraeyat.
My answers below:

1- You cannot add nodes to the mesh (at least, not easily). You can «glue» a node to one of the elements of the mesh (see some examples [here](https://github.com/xcfem/xc/tree/master/verif/tests/constraints)).
2- XC doesn't check for overlapped or ill defined multi-block topologies. The surfaces that overlap will have fully independent meshes.
3- You can iterate through the nodes of a line, or a surface. You can also search for its nearest node to a given position.
4- XC also uses the Wood-Armer method to check RC slabs. There is an example [here](https://github.com/xcfem/xc/blob/master/verif/tests/postprocess/limit_state_checking/ehe08/test_shell_normal_stresses_uls_checking.py).

**Comment by ebrahimraeyat** [comment]
Excuse me for my delay answer:
1- It appears that gmsh can do it: [link](https://stackoverflow.com/questions/36338025/how-to-ensure-node-points-in-gmsh)
3- As I said, it is the conditions that two or more surfaces overlaps each others, as you said, maybe two point in same position, each from a surface. I think freecad FEM WB does a function for this purpose, I will find it.
4- Thanks, I will study it.

**Comment by lcpt** [comment]
Hi, @ebrahimraeyat

1- You can use gmsh alone to create your mesh and then create the corresponding nodes and elements inside XC.
3- You can also glue a node to an element (there are some minimal examples here: https://github.com/xcfem/xc/tree/master/verif/tests/constraints test_glue_node_to_element_*.py) if this is what you want.
